remote_mapper.go

```diff
@@ -0,0 +1,196 @@
+package influxdb
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/influxdb/influxdb/influxql"
+)
+
+const (
+	MAX_MAP_RESPONSE_SIZE = 1024 * 1024 * 1024
+)
+
+// RemoteMapper implements the influxql.Mapper interface. The engine uses the remote mapper
+// to pull map results from shards that only exist on other servers in the cluster.
+type RemoteMapper struct {
+	dataNodes []*DataNode
+	resp      *http.Response
+	results   chan interface{}
+	unmarshal influxql.UnmarshalFunc
+	complete  bool
+
+	Call            string   `json:",omitempty"`
+	Database        string   `json:",omitempty"`
+	MeasurementName string   `json:",omitempty"`
+	TMin            int64    `json:",omitempty"`
+	TMax            int64    `json:",omitempty"`
+	SeriesIDs       []uint32 `json:",omitempty"`
+	ShardID         uint64   `json:",omitempty"`
+	Filters         []string `json:",omitempty"`
+	WhereFields     []*Field `json:",omitempty"`
+	SelectFields    []*Field `json:",omitempty"`
+	SelectTags      []string `json:",omitempty"`
+	Limit           int      `json:",omitempty"`
+	Offset          int      `json:",omitempty"`
+	Interval        int64    `json:",omitempty"`
+	ChunkSize       int      `json:",omitempty"`
+}
+
+// Responses get streamed back to the remote mapper from the remote machine that runs a local mapper
+type MapResponse struct {
+	Err       string `json:",omitempty"`
```

Comment:
You're talking about the fields in `RemoteMapper`? Those all need to be exported so they can be serialized and sent to the remote data node.